backend/api/Database/Models/MissionTask.cs

```diff
@@ -0,0 +1,136 @@
+            set
+            {
+                _taskStatus = value;
+                if (IsCompleted && EndTime is null)
+                    EndTime = DateTimeOffset.UtcNow;
+
+                if (_taskStatus is TaskStatus.InProgress && StartTime is null)
+                    StartTime = DateTimeOffset.UtcNow;
+            }
+        }
+
+        private bool IsCompleted =>
+            _taskStatus
+                is TaskStatus.Cancelled
+                    or TaskStatus.Successful
+                    or TaskStatus.Failed
+                    or TaskStatus.PartiallySuccessful;
+
+        public DateTimeOffset? StartTime { get; private set; }
+
+        public DateTimeOffset? EndTime { get; private set; }
+
+        public IList<Inspection> Inspections { get; set; }
+
+        public MissionTask() { }
+
+        public MissionTask(EchoTag echoTag, Position tagPosition)
+        {
+            Inspections = echoTag.Inspections
+                .Select(inspection => new Inspection(inspection))
+                .ToList();
+            EchoTagLink = echoTag.URL;
+            TagId = echoTag.TagId;
+            InspectionTarget = tagPosition;
+            RobotPose = echoTag.Pose;
+            EchoPoseId = echoTag.PoseId;
+            TaskOrder = echoTag.PlanOrder;
+        }
+
+        public void UpdateWithIsarInfo(IsarTask isarTask)
+        {
+            UpdateTaskStatus(isarTask.TaskStatus);
+            foreach (var inspection in Inspections)
+            {
+                var correspondingStep = isarTask.Steps.Single(
+                    step =>
+                        step.IsarStepId.Equals(
+                            inspection.IsarStepId.ToString(),
+                            StringComparison.Ordinal
+                        )
+                );
+                inspection.UpdateWithIsarInfo(correspondingStep);
+            }
+        }
+
+        public void UpdateTaskStatus(IsarTaskStatus isarStatus)
+        {
+            TaskStatus = isarStatus switch
+            {
+                IsarTaskStatus.NotStarted => TaskStatus.NotStarted,
+                IsarTaskStatus.InProgress => TaskStatus.InProgress,
+                IsarTaskStatus.Successful => TaskStatus.Successful,
+                IsarTaskStatus.PartiallySuccessful => TaskStatus.PartiallySuccessful,
+                IsarTaskStatus.Cancelled => TaskStatus.Cancelled,
+                IsarTaskStatus.Paused => TaskStatus.Paused,
+                IsarTaskStatus.Failed => TaskStatus.Failed,
+                _ => throw new ArgumentException($"ISAR Task status '{isarStatus}' not supported")
+            };
+        }
+
+#nullable enable
+        public Inspection? GetInspectionByIsarStepId(string isarStepId)
+        {
+            return Inspections.FirstOrDefault(
+                inspection => inspection.IsarStepId.Equals(isarStepId, StringComparison.Ordinal)
+            );
+        }
+
+#nullable disable
+    }
+
+    public enum TaskStatus
```

Comment:
Is this used outside the task or could it make sense to call it Status?

Reply:
Agreed to keep enum name and change variable name, since the enum is available from outside the class